app/controllers/events_controller.rb

```diff
@@ -27,6 +27,36 @@ def index
     end
   end
 
+  # GET /events/calendar
+  # GET /events/calendar.js
+  def calendar
+    # set a higher limit so we ensure to have enough results to fill a month
+    params[:per_page] ||= 200
+    @start_date = Date.parse(params.fetch(:start_date, Date.today.to_s)).beginning_of_month
+
+    set_params
+
+    # override @facet_params to get only events relevant for the current month view
+    @facet_params[:running_during] = "#{Date.today.beginning_of_day}/#{@start_date + 1.month}"
+    fetch_resources
+    events_set = @events
+
+    @facet_params[:running_during] = "#{@start_date}/#{@start_date + 1.month}"
+    fetch_resources
+    events_set += @events
+
```

Comment:
Right now it is done so that we make sure that the relevant events after the current day are not left out in favor of the past events if there are a lot of them. Looking for a better solution